File: roles/common/files/vmtools.pl
#!/usr/bin/perl -w
# install vm tools - rowlando -  rowland dot ono at gmail dot com 
# Copyright Centrality Ltd 14/05/2014
################################i################################
use strict;
use 5.10.0;
use File::Copy;

my $source = "VMwareTools-9.0.11-1743336.tar.gz";
my $dest = "/tmp";


say <<_WP_;

Before we begin, please ensure that you have installed VMWare Tools in the vApp VM Menu.

_WP_

# pause here for a bit so that the above warning is seen
sleep 2;

say "Starting install";

# change to the root dir
chdir "/root" or die "$!\n";

# check and create tools dir if 
# it does not exist
if ( ! -d "tools" ) {
    say "tools does not exist - creating it";
}
mkdir "tools", 0755;
say "tools directory created";

# mount tools dir to the dvd device
system ("mount", "/dev/sr0", "tools");

# check if the mount exists
open(F, "/etc/mtab") or die "$!\n";
my @string = (<F>);
close F;
my $f = grep /sr0/, @string;

# if the mount exists we are good to go
if ( $f ) {
    say "Ok - found $f";
}
# if the mount is missing, stop here and warn.
else {
    say <<_TOOLS; 
/root/tools not mounted - please check that you have installed VMWare Tools in the vApp VM Menu first.
    
_TOOLS
    exit 1;
}

# if we reach here, copy the files to the /tmp,
# uncompress them and install
chdir "tools" or die "$!\n";
say "Copying $source to $dest";
copy ($source, $dest) or die "File cannot be copied.\n";
chdir "/tmp";
say "uncompressing ${source}..";
system ("tar", "xvfz", "$source");
chdir "vmware-tools-distrib";
system ("./vmware-install.pl");

# unmount /root/tools dir
system ("umount", "/root/tools");

# delete /root/tools dir
rmdir "/root/tools" or die "$!\n";

File: roles/common/files/vmtools.py
#!/usr/bin/env python
# install vmware tools
# rowlando - 16/05/2014 rowland dot ono at gmail dot com
########################################################

import os
import sys
import time
import errno
import shutil
import subprocess

dest = "/tmp/"
#source = "VMwareTools-9.0.10-1481436.tar.gz"

print
print "Ensure that VMWare Tools is installed from the vApp VM Menu"
time.sleep(1)
print

print "Starting install"

try:
    os.chdir("/root")
except OSError:
    print "The directory does not exist. Exiting"
    exit
else:
    print "Changing to /root directory"

if not os.path.exists("/root/tools"):
    print "tools directory does not exist - creating it"
    os.mkdir ("tools", 0755)
    print "tools directory created"
else:
    print "tools directory exists - good"


subprocess.call("mount" +  " /dev/sr0" + " tools", shell=True)
m = open("/etc/mtab", "r")
lines = m.read()
result = lines.find('sr0')
if result == -1:
    print "/root/tools not mounted - check that you have installed VMWare Tools in the vApp VM Menu first."
    exit 

os.chdir("tools")
print "Changing to tools directory"

# find the vmwaretools file whatever its name
# as the name changes with updates
for file in os.listdir("/root/tools"):
    if file.endswith(".gz"):
        source = file


print "Copying " + source, " to " +  dest 

try:
    shutil.copy (source, dest)
except shutil.Error as e:
    print('Error: %s' % e)
except IOError as e:
    print('Error %s' % e.strerror)

os.chdir("/tmp")
print "Changing to /tmp directory"
print
print "Uncompressing" + ' source' 

subprocess.call("tar" +  " xvfz " +  source, shell=True)
os.chdir("vmware-tools-distrib")

subprocess.call("./vmware-install.pl", shell=True)

print "Cleaning up"
subprocess.call("umount" +  " /root/tools", shell=True)

if os.path.isfile(dest + source):
    os.remove(dest + source)
    print "Removed %s" % source, "from /tmp" 
else:
    print "Error: file not available for removal %s" % dest + source
